docs: enhanced workflow best practices for revision management (#1101)
mia-platform/documentation: docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

This workflow enables users to review and export generated configurations before these are even deployed to the cluster.  
This feature is available from the [Deploy Page](/development_suite/deploy/overview.md#export-and-review-configuration-files) after selecting the revision to deploy.

## Best Practices

> With the **Enhanced Project Workflow** paradigm shift, working with the Console may require for some change habits change; this section will present you some best practices that may help with such change.

Reviewer: ```suggestion
With the **Enhanced Project Workflow** paradigm shift, working with the Console may require some habit changes; this section presents some best practices that may help with such change.
```

### Revision management

> Working with revisions in the Enhanced Workflow is not that different from working with branches in the Standard Workflow; revisions have their own history and can be deployed on different environments therefore patterns such as **Git Flow** or **Trunk-based development** can be applied to the **Enhanced Project Workflow** as well.

Reviewer: ```suggestion
Working with revisions in the Enhanced Workflow is not that different from working with branches in the Standard Workflow; revisions have their own history and can be deployed on different environments, therefore patterns such as **Git Flow** or **Trunk-based development** can be applied to the **Enhanced Project Workflow** as well.
```

#### Trunk-based

> If you are **working in a small team**, in general, we advise to use **trunk-based development**, with a single revision that represents your Project state-of-the-art that gets versioned everytime a release is needed.

Reviewer: ```suggestion
If you are **working in a small team**, in general, we advise using **trunk-based development**, with a single revision that represents your Project's state-of-the-art that gets versioned every time a release is needed.
```

> Create new revisions any time you need to make experiments or you are unsure about the changes you need to make but [*remember to merge*](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) them in the main trunk as soon as possible to prevent headaches in reconciling diverging revisions.

Reviewer 2: ```suggestion
Create new revisions whenever you need to experiment or are unsure of the changes to be made, but [*remember to merge*](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) them in the main trunk as soon as possible to prevent headaches in reconciling diverging revisions.
```

> Create new revision any time you need to make experiments or you are unsure about the changes you need to make but [*remember to merge*](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) them in the main trunk as soon as possible to prevent headaches in reconciling diverging revisions.

Reviewer: ```suggestion
Create new revisions any time you need to make experiments or you are unsure about the changes you need to make but [*remember to merge*](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) them in the main trunk as soon as possible to prevent headaches in reconciling diverging revisions.
```

:::tip

> When a version is deployed a tag is created on Git. Such tag has a different name based on the target environment making it possible to verify configuration changes between different deployment of the same version!

Reviewer: ```suggestion
When a version is deployed a tag is created on Git. Such tag has a different name based on the target environment making it possible to verify configuration changes between different deployments of the same version!
```

:::

#### Git Flow

> If you are working in **larger teams** or need to make more **structured change reviews**, use instead **Git Flow**, create a revision for each environment your project has. The daily work will be done in the revision associated with the "lower" environment;

Reviewer: ```suggestion
If you are working in **larger teams** or need to make more **structured change reviews**, use **Git Flow** instead and create a revision for each environment your project has. The daily work will be done in the revision associated with the "lower" environment;
```

> each time changes can be promoted to a "higher" environment use the [merge](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) feature to bring changes to the revision associated with it.

Reviewer: ```suggestion
whenever changes can be promoted to a "higher" environment, use the [merge](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) feature to bring those changes to the revision associated with it.
```

:::tip

> In a project with two environments, development and production, make sure to have at least two revisions.

Reviewer: ```suggestion
In a project with two environments (e.g. development and production), make sure to have at least two revisions.
```

> Developers will make changes in the `development` revision and deploy such revision in the development environment; to promote changes in the production environment, use the [merge](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) feature to bring them in the correct revision and finally deploy it to the production environment.

Reviewer: ```suggestion
Developers will make changes in the `development` revision and deploy such revision in the development environment; to promote changes to the production environment, use the [merge](/development_suite/api-console/api-design/merge_collaboration.md#how-to-perform-a-merge-of-configurations) feature to incorporate them into the correct revision and finally deploy it to the production environment.
```
